src/Microsoft.AspNet.Mvc.Core/DefaultControllerFactory.cs

```diff
@@ -38,8 +38,19 @@ public DefaultControllerFactory(IControllerActivator controllerActivator)
             _getPropertiesToActivate = GetPropertiesToActivate;
         }
 
+        /// <summary>
+        /// The <see cref="IControllerActivator"/> used to create a controller.
+        /// </summary>
+        protected IControllerActivator ControllerActivator
```

Comment:
The factory caches things, consequently it's a singleton. Everything consumed by a singleton ends up having a singleton lifetime.